### PR TITLE
Makes catslugs slightly less annoying

### PR DESCRIPTION
**Pull request by Very-Soft**
Now they ONLY respond to psps, rather than everything except psps except when they are kicking your ass.

Also makes it so simplemobs get up to kick your ass if you attack them when they are resting.

They also!
Can be picked up now.
They also may repeat what you say, or ask very important questions to you.

### CODE REVIEW

File: code/modules/mob/living/simple_mob/subtypes/animal/alien animals/catslug.dm [hunk 1]
```diff
@@ -178,19 +189,32 @@
...
 /datum/ai_holder/simple_mob/melee/evasive/catslug/on_hear_say(mob/living/speaker, message)
+	if(holder.client)
+		return
+	if(!speaker.client)
+		return
+	if(findtext(message, "psps") && stance == STANCE_IDLE)
+		set_follow(speaker, follow_for = 5 SECONDS)
+
+/mob/living/simple_mob/vore/alienanimals/catslug/horrible
+/datum/ai_holder/simple_mob/melee/evasive/catslug/horrible/on_hear_say(mob/living/speaker, message)	//this was an accident originally but it was very funny so here you go
```
Comment (Arokha) [inline code/modules/mob/living/simple_mob/subtypes/animal/alien animals/catslug.dm:217]: You didn't actually set the holder type on the horrible catslug to the horrible type of ai

File: code/modules/mob/living/simple_mob/subtypes/animal/alien animals/catslug.dm [hunk 2]
```diff
@@ -178,19 +189,32 @@
...
+		addtimer(CALLBACK(src, .proc/consider_awakening), rand(1 MINUTE, 5 MINUTES), TIMER_UNIQUE|TIMER_OVERRIDE|TIMER_STOPPABLE)
+	else 
+		return ..()
+
 
 /datum/ai_holder/simple_mob/melee/evasive/catslug/on_hear_say(mob/living/speaker, message)
+	if(holder.client)
+		return
+	if(!speaker.client)
+		return
```
Comment (Arokha) [inline code/modules/mob/living/simple_mob/subtypes/animal/alien animals/catslug.dm:212]: Could combine if you want

File: code/modules/ai/ai_holder_targeting.dm [hunk 3]
```diff
@@ -252,6 +252,10 @@
 		on_attacked(attacker)
 		return FALSE
 
+	if(holder.resting)	// I can't kill someone while I'm laying down!
+		ai_log("react_to_attack() : AI is resting. Getting up up.", AI_LOG_TRACE)
```
Comment (Arokha) [inline code/modules/ai/ai_holder_targeting.dm:256]: up up

File: code/modules/mob/living/simple_mob/subtypes/animal/alien animals/catslug.dm [hunk 4]
```diff
@@ -137,6 +137,17 @@
...
+	if(stat == DEAD)
+		return ..()
+	if(M.a_intent == I_HELP && resting)
+		M.visible_message("<span class='notice'>\The [M.name] shakes \the [src] awake from their nap.</span>","<span class='notice'>You shake \the [src] awake!</span>")
+		playsound(src, 'sound/weapons/thudswoosh.ogg', 50, 1, -1)
+		lay_down()
+		ai_holder.go_wake()
+		return
+	return ..()
+
```
Comment (Arokha) [inline code/modules/mob/living/simple_mob/subtypes/animal/alien animals/catslug.dm:150]: Could be rewritten more simply as:
```suggestion
/mob/living/simple_mob/vore/alienanimals/catslug/attack_hand(mob/living/carbon/human/M as mob)
	if(stat != DEAD && M.a_intent == I_HELP && resting)
		M.visible_message("<span class='notice'>\The [M.name] shakes \the [src] awake from their nap.</span>","<span class='notice'>You shake \the [src] awake!</span>")
		playsound(src, 'sound/weapons/thudswoosh.ogg', 50, 1, -1)
		lay_down()
		ai_holder.go_wake()
		return
	else
		return ..()
```

**Comment by Very-Soft** [comment]
Did all the things you suggested!

**Pull request merged**